File: Classes/EVEOnlineAPI/Corporation/EVEMemberSecurityLog.m
//
//  EVEMemberSecurityLog.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 1/7/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "EVEMemberSecurityLog.h"

@implementation EVEMemberSecurityLogRolesItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"roleID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"roleName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	
	return scheme;
}

@end


@implementation EVEMemberSecurityLogRoleHistoryItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"changeTime":@{@"type":@(EVEXMLSchemePropertyTypeDate)},
				   @"characterID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"characterName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"issuerID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"issuerName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"roleLocationType":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"oldRoles":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityLogRolesItem class]},
				   @"theNewRoles":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityLogRolesItem class]}};
	
	return scheme;
}

@end


@implementation EVEMemberSecurityLog

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"roleHistory":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityLogRoleHistoryItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEMemberSecurityLog.swift
//
//  EVEMemberSecurityLog.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEMemberSecurityLogRoleHistoryItem: EVEObject {
	public var changeTime: Date = Date.distantPast
	public var characterID: Int64 = 0
	public var characterName: String = ""
	public var issuerID: Int64 = 0
	public var issuerName: String = ""
	public var roleLocationType: String = ""
	public var oldRoles: [EVEMemberSecurityRole] = []
	public var theNewRoles: [EVEMemberSecurityRole] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"changeTime":EVESchemeElementType.Date(elementName:nil, transformer:nil),
		"characterID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
		"characterName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		"issuerID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
		"issuerName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		"roleLocationType":EVESchemeElementType.String(elementName:nil, transformer:nil),
		"oldRoles":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"theNewRoles":EVESchemeElementType.Rowset(elementName: "newRoles", type: EVEMemberSecurityRole.self, transformer: nil),
		]
	}
}


public class EVEMemberSecurityLog: EVEResult {
	public var roleHistory: [EVEMemberSecurityLogRoleHistoryItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"roleHistory":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityLogRoleHistoryItem.self, transformer: nil),
		]
	}
}

